python/cuml/common/base.pyx

```diff
@@ -365,7 +365,7 @@ class Base(TagsMixin,
             return {'preserves_dtype': [self.dtype]}
         return {}
 
-    def _set_nvtx_annotations(self):
+    def set_nvtx_annotations(self):
```

Comment:
I like this approach, I was thinking perhaps we could check for an environment variable in the init and call this so as to have an easy library wide way of enabling nvtx around, what do you think?

Reply:
Sure, I can do this.